tools/format.py

```diff
@@ -26,7 +28,10 @@
 #   * These third party python files shouldn't be formatted.
 #   * The tools directory has no subdirectories, so `glob()` is sufficient.
 # TODO(ry) Install yapf in third_party.
-run(["yapf", "-i"] + glob("tools/*.py") + find_exts("build_extra", ".py"))
+if find_executable('yapf'):
+    run(["yapf", "-i"] + glob("tools/*.py") + find_exts("build_extra", ".py"))
+else:
+    raise Exception("yapf not available!. install `pip install yapf`")
```

Comment:
I think printing and sys.exit might be preferable to a raise here. Since you don't really want the stack trace. Also `!.` is a little weird here. Maybe it would read better as:
> Unable to format python code: yapf not found. Install with ``pip install yapf``.

But this is aside from cwd!

Reply:
yes i agree. removing them from this pr